### PR TITLE
[msal-angular] Add additional interfaces to msal-guard

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR:
- Adds `CanActivateChild` and `CanLoad` to the guard
- Adds unit tests for `CanActivateChild` and `CanLoad`
- Updates msal-angular initialization doc and upgrade guide to reflect the above changes
- Adds `detail` component to angular11-sample-app to demonstrate use of `CanActivateChild`
- Adds lazy loading module to angular11-sample-app to demonstrate use of `CanLoad`

This PR is in response to issue: #1167.

### CODE REVIEW

File: lib/msal-angular/src/msal.guard.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { concatMap, catchError, map } from "rxjs/operators";
 import { Observable, of } from "rxjs";
 
 @Injectable()
-export class MsalGuard implements CanActivate {
+export class MsalGuard implements CanActivate, CanActivateChild {
```
Comment (sameerag) [inline lib/msal-angular/src/msal.guard.ts:17]: Is this broker related?

Comment (jasonnutter) [reply lib/msal-angular/src/msal.guard.ts:17]: No, this is related to child routes in Angular (e.g. contoso.com/product/123)

Comment (jo-arroyo) [reply lib/msal-angular/src/msal.guard.ts:17]: Nope! This is in response to an Angular issue from 1.x. I don't think it will affect broker.

File: change/@azure-msal-angular-221b39d9-8837-4a95-872a-f594ec63a6af.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jasonnutter) [inline change/@azure-msal-angular-221b39d9-8837-4a95-872a-f594ec63a6af.json:2]:
```suggestion
  "type": "prerelease",
```

Comment (jasonnutter) [reply change/@azure-msal-angular-221b39d9-8837-4a95-872a-f594ec63a6af.json:2]: Still needs to be `prerelease` while in alpha.

**Review by jasonnutter — APPROVED** [review]
Need to be fix change file, but otherwise looks good!

**Review by tnorling — APPROVED** [review]
